jsonargparse/_typehints.py

```diff
@@ -848,7 +848,7 @@ def adapt_typehints(
                 subtypehint = subtypehints[0 if is_ellipsis or not is_tuple else n]
                 val[n] = adapt_typehints(v, subtypehint, **adapt_kwargs)
             if is_tuple and len(val) == 0:
-                raise_unexpected_value("Expected a non-empty tuple", val)
```

Comment:
Don't remove the raise for empty tuple. This is correct and necessary. Only tuples with `...` should be allowed to be empty. But other tuples not. The code here should be:
```python
if is_tuple and len(val) == 0 and not is_ellipsis:
    raise_unexpected_value("Expected a non-empty tuple", val)
```

Reply:
Actually no. These two lines should be removed. Number of elements in the tuple are already handled by
```python
if is_tuple and not is_ellipsis and len(val) != len(subtypehints):
    raise_unexpected_value(f"Expected a tuple with {len(subtypehints)} elements", val)
```
And converting the empty list to a tuple would already be handled by
```python
if not serialize:
    val = tuple(val) if typehint_origin in {Tuple, tuple} else set(val)
```